activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -457,6 +455,15 @@ def each_local_configuration
         def local_database?(configuration)
           configuration["host"].blank? || LOCAL_HOSTS.include?(configuration["host"])
         end
+
+        def dump_ruby_schema(filename)
+          temp_file = Tempfile.new("temp_schema_dump.rb")
+          ActiveRecord::SchemaDumper.dump(ActiveRecord::Base.connection, temp_file)
```

Comment:
Seems like you could just use `File.atomic_write` here: https://api.rubyonrails.org/classes/File.html#method-c-atomic_write